docs/developer-guide/joins/partition-data.md

```diff
@@ -17,6 +17,10 @@ records based on the joining column. To ensure that records with the same
 join column are co-located on the same stream task, the join column must
 coincide with the column that the sources are partitioned by.
 
+Partitioning streams and tables is especially important for stateful or otherwise
+intensive queries. For more information, see
+[Parallelization](/operate-and-deploy/performance-guidelines/#parallelization).
```

Comment:
Checking my understanding - the parallelization docs say that the rule of thumb is twice as many partitions as nodes. Can customers easily see how many nodes they have? IIUC a basic deployment will have 4 nodes right, so 8 partitions? I think in conversation John and Almog recommended 6 partitions as a baseline (but we don't necessarily need to get into that). Either way might be nice to add a little blurb about a basic option depending on what you all think

Reply:
@lct45 Sounds good, wdyt @agavra?

Reply:
@lct45 I'm going to merge as-is, and we can chat about adding a basic-option blurb in the parallelization topic.